airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -123,38 +150,33 @@ def _configure_logger_level(self, logger: logging.Logger):
             logger.setLevel(logging.DEBUG)
 
     def _validate_source(self):
-        # Validates the connector manifest against the schema auto-generated from the low-code backend
-        full_config = {}
-        if "version" in self._source_config:
-            full_config["version"] = self._source_config["version"]
-        full_config["check"] = self._source_config["check"]
-        streams = [self._factory.create_component(stream_config, {}, False)() for stream_config in self._stream_configs()]
-        if len(streams) > 0:
-            full_config["streams"] = streams
-        declarative_source_schema = ConcreteDeclarativeSource.json_schema()
-
-        try:
-            validate(full_config, declarative_source_schema)
-        except ValidationError as e:
-            raise ValidationError("Validation against auto-generated schema failed") from e
-
-        # Validates the connector manifest against the low-code component json schema
-        manifest = self._source_config
-        if "type" not in manifest:
-            manifest["type"] = "DeclarativeSource"
-        manifest_transformer = ManifestComponentTransformer()
-        propagated_manifest = manifest_transformer.propagate_types_and_options("", manifest, {})
-
-        try:
-            raw_component_schema = pkgutil.get_data("airbyte_cdk", "sources/declarative/declarative_component_schema.yaml")
-            declarative_component_schema = yaml.load(raw_component_schema, Loader=yaml.SafeLoader)
-        except FileNotFoundError as e:
-            raise FileNotFoundError(f"Failed to read manifest component json schema required for validation: {e}")
-
-        try:
-            validate(propagated_manifest, declarative_component_schema)
-        except ValidationError as e:
-            raise ValidationError("Validation against json schema defined in declarative_component_schema.yaml schema failed") from e
+        if self.construct_using_pydantic_models:
+            # Validates the connector manifest against the low-code component schema defined in declarative_component_schema.yaml
+            try:
+                raw_component_schema = pkgutil.get_data("airbyte_cdk", "sources/declarative/declarative_component_schema.yaml")
+                declarative_component_schema = yaml.load(raw_component_schema, Loader=yaml.SafeLoader)
+            except FileNotFoundError as e:
+                raise FileNotFoundError(f"Failed to read manifest component json schema required for validation: {e}")
+
+            try:
+                validate(self._source_config, declarative_component_schema)
```

Comment:
nit: we should validate using the legacy validation until we delete the code (unless we're 110% confident they're equivalent)

Reply:
Why is that? I'm wondering because I feel like the earlier we enable this validation, the more feedback we will get to know if it behaves as we expect?

Reply:
We did validate it in part 1, but its trivial to leave it in and doesn't hurt to so I'll remove the conditional here, but keep the rest to gate construction.